src/core/api/thread_api.cpp

```diff
@@ -501,4 +501,20 @@ bool otThreadIsAnycastLocateInProgress(otInstance *aInstance)
 }
 #endif
 
+void otThreadLeaveNetwork(otInstance *aInstance)
+{
+    static constexpr otOperationalDatasetTlvs kEmptyDatasetTlvs = {};
+
+    SuccessOrAssert(otThreadSetEnabled(aInstance, false));
+    if (otIp6IsEnabled(aInstance))
+    {
+        SuccessOrAssert(otIp6SetEnabled(aInstance, false));
+    }
+    SuccessOrAssert(otDatasetSetActiveTlvs(aInstance, &kEmptyDatasetTlvs));
+    SuccessOrAssert(otDatasetSetPendingTlvs(aInstance, &kEmptyDatasetTlvs));
```

Comment:
These will delete the active/pending dataset in the `Settings` (non-volatile).

There are other things in settings, like `kKeyChildInfo`, `kKeyNetworkInfo` that are tied to the current network.

If we want to leave the network and potentially join a different one, we should clear everything in `Settings`.

Reply:
Also I think we should clear everything in RAM (used by OT stack) as well.